File: src/main/java/xyz/xpto/gerenciamento/domain/entities/Equipe.java
package xyz.xpto.gerenciamento.domain.entities;

import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@Table(name = "equipe")
public class Equipe {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "equipe_id_sequence")
    @SequenceGenerator(name = "equipe_id_sequence", sequenceName = "equipe_id_sequence", allocationSize = 1)
    @Column(name = "id")
    private long id;

    @Column(name = "nome", nullable = false)
    private String nome;

    @OneToMany(mappedBy = "equipe", targetEntity = Pessoa.class, orphanRemoval = true, cascade = CascadeType.ALL)
    private List<Pessoa> pessoas;

}

File: frontend/types/Equipe.ts
import type Pessoa from "./Pessoa"
import type Projeto from "./Projeto"

type EquipeParams = {
	id?: number
	nome?: string
	pessoas?: Pessoa[]
	projetos?: Projeto[]
}

export default class Equipe {
	public id: number;
	public nome: string;
	public pessoas: Pessoa[];
	public projetos: Projeto[];

	constructor({ id, nome, pessoas, projetos }: EquipeParams
	) {
		this.id = id || 0
		this.nome = nome || 'Sem nome'
		this.pessoas = pessoas || []
		this.projetos = projetos || []
	}
}